crates/core/src/database.rs

```diff
@@ -47,18 +47,19 @@ pub struct Database {
     pub db_path: String,
     #[cfg(feature = "replication")]
     pub replication_ctx: Option<ReplicationContext>,
+    pub readonly: bool,
 }
 
 impl Database {
     /// Open a local database file.
-    pub fn open<S: Into<String>>(db_path: S) -> Result<Database> {
+    pub fn open<S: Into<String>>(db_path: S, readonly:bool) -> Result<Database> {
```

Comment:
We should do a `open_with_flags()` associate function like Rusqlite does for this instead of adding a boolean to `open()` that everybody has to pass.